ndbench-dynamodb-plugins/src/main/java/com/netflix/ndbench/plugin/dynamodb/DynamoDBProgrammaticKeyValue.java

```diff
@@ -0,0 +1,261 @@
+    private long publishingInterval;
+    private Dimension tableDimension;
+
+    @Override
+    public void init(DataGenerator dataGenerator) {
+        dynamoDBKeyValue.init(dataGenerator);
+
+        //get dynamodb client from the encapsulated DynamoDBKeyValue
+        AmazonDynamoDB dynamoDB = dynamoDBKeyValue.getDynamoDB();
+
+        //prerequisite data from configuration
+        String tableName = config.getTableName();
+        String partitionKeyName = config.getAttributeName();
+        long rcu = Long.parseLong(config.getReadCapacityUnits());
+        long wcu = Long.parseLong(config.getWriteCapacityUnits());
+        this.publishingInterval = config.getHighResolutionMetricsPublishingInterval();
+        this.tableDimension = new Dimension().withName("TableName").withValue(tableName);
+
+        //setup control plane operations
+        this.describeLimits = new DescribeLimits(dynamoDB, tableName, partitionKeyName);
+        this.createTable = new CreateDynamoDBTable(dynamoDB, tableName, partitionKeyName, rcu, wcu);
+        this.deleteTable = new DeleteDynamoDBTable(dynamoDB, tableName, partitionKeyName);
+        logger.info("Creating table programmatically");
+        TableDescription td = createTable.get();
+        logger.info("Table Description: " + td.toString());
+        final DescribeLimitsResult limits = describeLimits.get();
+        final Integer targetWriteUtilization = Integer.parseInt(config.getTargetWriteUtilization());
+        final Integer targetReadUtilization = Integer.parseInt(config.getTargetReadUtilization());
+        dynamoDBAutoscalingConfigurer.setupAutoscaling(rcu, wcu, config.getTableName(), limits, targetWriteUtilization,
+                targetReadUtilization);
+
+        // build cloudwatch client
+        AmazonCloudWatchClientBuilder cloudWatchClientBuilder = AmazonCloudWatchClientBuilder.standard();
+        cloudWatchClientBuilder.withCredentials(awsCredentialsProvider);
+        if (StringUtils.isNotEmpty(this.config.getRegion())) {
+            cloudWatchClientBuilder.withRegion(Regions.fromName(config.getRegion()));
+        }
+        cloudWatch = cloudWatchClientBuilder.build();
+        putMetricAlarm = new PutMetricAlarmOperation(cloudWatch);
+        putMetricData = new PutMetricDataOperation(cloudWatch);
+
+        if (config.publishHighResolutionConsumptionMetrics()) {
+            logger.info("Initializing CloudWatch reporter");
+            checkAndInitCloudwatchReporter();
+        }
+
+        if (config.alarmOnHighResolutionConsumptionMetrics()) {
+            Preconditions.checkState(config.publishHighResolutionConsumptionMetrics());
+            //create a high resolution alarm for consuming 80% or more RCU of provisioning
+            createHighResolutionAlarm(
+                    "ndbench/DynamoDB/RcuConsumedAlarm",
+                    ND_BENCH_DYNAMO_DB_CONSUMED_RCU,
+                    0.8 * Double.valueOf(config.getReadCapacityUnits()));
```

Comment:
Do we want the 0.8 to be final variable in the top or some configuration parameter?

Reply:
ill pull it into configuration